### PR TITLE
Add preserve data in the browser function

### PR DESCRIPTION
**Pull request by MostafaShahat2021**
⭐In this branch we:
 - Add preserve data in the browser function
 - Create a single JavaScript object with all the data from the entire form and save it in local storage

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -198,4 +204,37 @@ const formCheck = () => {
...
+body.onload = () => {
+  if (recievingData) {
+    const dataSave = JSON.parse(recievingData);
+    nameValue.value = dataSave.name;
+    emailInput.value = dataSave.email;
+    msg.value = dataSave.message;
+  }
+};
+
+document.querySelectorAll('input').forEach((input) => {
```
Comment (MaganaAsiati) [inline main.js]: - You did a great job by using  a queryselectorAll to select  your  all  input but for the message, you didn't use the input you use the textarea , your name and email are saved correctly in the local storage as required but the massage  is not saving as it showing bellow.   :-1: :smiley:  :smiley: 

image
  
- [ ] Kindly make sure you fix this error, the simple to fix this error is to add textarea in your queryselectorAll().check below the format. ::smiley:  :smiley: :100: :+1:  
`document.querySelectorAll('input, textarea').forEach((input) => {`

**Review by MaganaAsiati — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️♻️♻️ 

Hi @MostafaShahat2021 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlonzoLeroneThisCloseGIF

## Highlights
✅ No linter error
✅ README is professional

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: main.js (hunk 1 above)
Comment (MostafaShahat2021) [reply main.js]: @MaganaAsiati Thanks a lot 🙌 for your efforts in reviewing my code, I really appreciate it😊.

**Review by AshakaE — APPROVED** [review]
Hi @MostafaShahat2021 

### Highlights ✨ 

- Pull request 👌 
- Git flow ✅ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MostafaShahat2021** [comment]
@AshakaE Thank you so much 🙌 for your efforts in reviewing my code.

**Pull request merged**